core/src/smartcontracts/isi/account.rs

```diff
@@ -611,4 +614,33 @@ pub mod query {
             .map_err(Into::into)
         }
     }
+
+    impl ValidQuery for FindAccountsWithPermissionTokenDefinition {
```

Comment:
we don't need yet another query. You can achieve the same using `FindAllAccounts` + filter

Reply:
As I understand we don't have a way to attach filter to a query in WASM. So I need to do it on WASM side. Where is the edge of what we should put into Iroha and what into client?
Although validator is not a basic client and this query has a big chance to be popular in this scenarios

Reply:
> Where is the edge of what we should put into Iroha and what into client?

That's an interesting question. IMO there are 2 categories of queries:

1. queries executed by a client (can only be done through ISI)
2. queries executed by a smart contract (can be done through ISI or custom `extern` function)

Now for the 2nd category. What I'm objecting to is that we've started expanding ISI queries just for the purposes of **internal execution** of validator. No client will likely ever execute these queries. IMO, that's not ISI should be structured, rather the set of queries should be expanded according to the needs of the client. Internally used queries should be `extern` functions exposed by the host just like any other entrypoint

We should get more opinions here

Reply:
I can't see much difference between creating `extern` function and creating new query. But with query this becomes usefull for basic clients as well, so double kill, because this shorthand query might be valuable for them as well

Reply:
- Created #3720